zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Notebook.java

```diff
@@ -183,6 +183,34 @@ public String exportNote(String noteId) throws IOException, IllegalArgumentExcep
   }
 
   /**
+   *
+   * @param url - the ipfs url or hash to get note from
+   * @param subject
+   * @return note
+   * @throws IOException
+   */
+  public Note importNote(String url, AuthenticationInfo subject) throws IOException {
+    Note newNote = null;
+    Note oldNote = notebookRepo.getNoteFromUrl(url, subject);
+    if (oldNote != null) {
+      try {
+        newNote = createNote(subject);
+        newNote.setName(oldNote.getName());
+        List<Paragraph> paragraphs = oldNote.getParagraphs();
+        for (Paragraph p : paragraphs) {
+          newNote.addCloneParagraph(p);
+        }
+        newNote.persist(subject);
+      } catch (IOException e) {
+        logger.error(e.toString(), e);
```

Comment:
Instead of `e.toString()` we could have something like "Importing note from " + url+ " failed". The error message will still be there, as we propaget `e`.